pkg/agent/agent_windows.go

```diff
@@ -204,6 +294,90 @@ func (i *Initializer) prepareOVSBridgeOnHNSNetwork() error {
 	return nil
 }
 
+func (i *Initializer) prepareOVSConfigForVM() error {
+	klog.InfoS("Performing OVS configuration", "vmInterface", i.vmInterface)
+	hostIfName := i.vmInterface
+	uplinkIfName := util.GenUplinkInterfaceName(hostIfName)
+	ovsPorts, err := i.ovsBridgeClient.GetPortList()
+	if err != nil {
+		klog.Errorf("Failed to list OVS ports: %v", err)
+		return err
+	} else {
+		for index := range ovsPorts {
+			port := &ovsPorts[index]
+			if port.Name == hostIfName {
+				klog.InfoS("Uplink and host configuration exists in OVS")
+				return nil
+			}
+		}
+	}
+
+	success := false
+	uplinkExternalIDs := map[string]interface{}{
+		interfacestore.AntreaInterfaceTypeKey: interfacestore.AntreaUplink,
+	}
+	// Create uplink port on OVS.
+	uplinkUUID, err := i.ovsBridgeClient.CreatePort(uplinkIfName, uplinkIfName, uplinkExternalIDs)
+	if err != nil {
+		klog.ErrorS(err, "Failed to create uplink port on OVS", "uplink", uplinkIfName)
+		return err
+	}
+
+	defer func() {
+		if !success {
+			klog.InfoS("Exitting!!  delete port on OVS", "uplinkUUID", uplinkUUID)
+			i.ovsBridgeClient.DeletePort(uplinkUUID)
+		}
+	}()
+
+	// Query the uplink port to check if its created
+	uplinkOFPort, err := i.ovsBridgeClient.GetOFPort(uplinkIfName, false)
+	if err != nil {
+		klog.ErrorS(err, "Failed to get uplink ofport", "uplink", uplinkIfName)
+		return err
+	}
+	klog.InfoS("Added uplink port on OVS", "port", uplinkOFPort)
+
+	hostIF, error := net.InterfaceByName(hostIfName)
+	if error != nil {
+		klog.ErrorS(err, "Failed to get InterfaceByName", "hostIfName", hostIfName)
+		return error
+	}
+
+	attachInfo := map[string]interface{}{
```

Comment:
Is it easier if we public the function to generate the attachInfo of an OVS internal port for the ExternalNode interface? Then we don't need to know the details of these external_ids keys.

Reply:
It's better to use `getOVSAttachInfo` from externalnode package. Should this function be in externalnode package?